cmd/shim/main.go

```diff
@@ -286,22 +286,30 @@ func shim() int {
 	outWriter := io.MultiWriter(stdoutFile, os.Stdout)
 	errWriter := io.MultiWriter(stderrFile, os.Stderr)
 
-	// Only initialize scrubbing logic if secrets are set:
 	if len(secretsToScrub.Envs) == 0 && len(secretsToScrub.Files) == 0 {
 		cmd.Stdout = outWriter
 		cmd.Stderr = errWriter
 	} else {
 		envToScrub := os.Environ()
-		scrubOutWriter, err := NewSecretScrubWriter(outWriter, currentDirPath, shimFS, envToScrub, secretsToScrub)
+		stdoutPipe, err := cmd.StdoutPipe()
 		if err != nil {
 			panic(err)
 		}
-		cmd.Stdout = scrubOutWriter
-		scrubErrWriter, err := NewSecretScrubWriter(errWriter, currentDirPath, shimFS, envToScrub, secretsToScrub)
+		scrubOutReader, err := NewSecretScrubReader(stdoutPipe, currentDirPath, shimFS, envToScrub, secretsToScrub)
 		if err != nil {
 			panic(err)
 		}
-		cmd.Stderr = scrubErrWriter
+		go io.Copy(outWriter, scrubOutReader)
+
+		stderrPipe, err := cmd.StderrPipe()
```

Comment:
@matiasinsaurralde  I think the test failures are due to use of `cmd.StdoutPipe` and `cmd.StderrPipe`. The [docs](https://pkg.go.dev/os/exec#Cmd.StderrPipe) for those say:
> It is thus incorrect to call Wait before all reads from the pipe have completed. For the same reason, it is incorrect to use Run when using StderrPipe. See the StdoutPipe example for idiomatic usage. 

I was able to repro the failures consistently when pulling this PR locally, I was no longer able to repro w/ this (very rough patch):
```
diff --git a/cmd/shim/main.go b/cmd/shim/main.go
index ecdc8567..09972d94 100644
--- a/cmd/shim/main.go
+++ b/cmd/shim/main.go
@@ -291,21 +291,17 @@ func shim() int {
                cmd.Stderr = errWriter
        } else {
                envToScrub := os.Environ()
-               stdoutPipe, err := cmd.StdoutPipe()
-               if err != nil {
-                       panic(err)
-               }
-               scrubOutReader, err := NewSecretScrubReader(stdoutPipe, currentDirPath, shimFS, envToScrub, secretsToScrub)
+               outR, outW := io.Pipe()
+               cmd.Stdout = outW
+               scrubOutReader, err := NewSecretScrubReader(outR, currentDirPath, shimFS, envToScrub, secretsToScrub)
                if err != nil {
                        panic(err)
                }
                go io.Copy(outWriter, scrubOutReader)
 
-               stderrPipe, err := cmd.StderrPipe()
-               if err != nil {
-                       panic(err)
-               }
-               scrubErrReader, err := NewSecretScrubReader(stderrPipe, currentDirPath, shimFS, envToScrub, secretsToScrub)
+               errR, errW := io.Pipe()
+               cmd.Stderr = errW
+               scrubErrReader, err := NewSecretScrubReader(errR, currentDirPath, shimFS, envToScrub, secretsToScrub)
                if err != nil {
                        panic(err)
                }
```